File: Optimised/validator.py
"""
validator.py - Validator component (Optimised).
Functionally identical to the baseline — validation logic does not change.
"""
from .metrics import tracker


class Validator:
    @tracker.track
    def validateFormat(self, data: dict) -> bool:
        required_fields = ["title", "content", "author"]
        return all(field in data and bool(data[field]) for field in required_fields)

File: Optimised/src/Validator.java
public class Validator {

    public ValidationResult validateFormat(String[] data) {
        MetricTracker.record("Validator.validateFormat");
        if (data == null) {
            return new ValidationResult(false, "Data is null");
        }
        if (data.length == 0) {
            return new ValidationResult(false, "Research format is incorrect");
        }
        return new ValidationResult(true, null);
    }
}
